Framework/HistogramData/test/HistogramTest.h

```diff
@@ -837,6 +851,72 @@ class HistogramTest : public CxxTest::TestSuite {
     hist.setCountStandardDeviations(data1);
     TS_ASSERT_THROWS(hist.setSharedE(data2), std::logic_error);
   }
+
+  void test_yMode() {
+    Histogram hist1(Histogram::XMode::Points, Histogram::YMode::Counts);
+    TS_ASSERT_EQUALS(hist1.yMode(), Histogram::YMode::Counts);
+    Histogram hist2(Histogram::XMode::Points, Histogram::YMode::Frequencies);
+    TS_ASSERT_EQUALS(hist2.yMode(), Histogram::YMode::Frequencies);
+  }
+
+  void test_yMode_Uninitialized() {
+    Histogram hist(Points(1));
+    TS_ASSERT_EQUALS(hist.yMode(), Histogram::YMode::Uninitialized);
+  }
+
+  void test_yMode_initialized_by_setCounts() {
+    Histogram h(Points(2));
+    h.setCounts(2);
+    TS_ASSERT_EQUALS(h.yMode(), Histogram::YMode::Counts);
+  }
+
+  void test_yMode_initialized_by_setCountStandardDeviations() {
+    Histogram h(Points(2));
+    h.setCountStandardDeviations(2);
+    TS_ASSERT_EQUALS(h.yMode(), Histogram::YMode::Counts);
+  }
+
+  void test_yMode_initialized_by_setCountVariances() {
+    Histogram h(Points(2));
+    h.setCountVariances(2);
+    TS_ASSERT_EQUALS(h.yMode(), Histogram::YMode::Counts);
+  }
+
+  void test_yMode_initialized_by_setFrequencies() {
+    Histogram h(Points(2));
+    h.setFrequencies(2);
+    TS_ASSERT_EQUALS(h.yMode(), Histogram::YMode::Frequencies);
+  }
+
+  void test_yMode_initialized_by_setFrequencyStandardDeviations() {
+    Histogram h(Points(2));
+    h.setFrequencyStandardDeviations(2);
+    TS_ASSERT_EQUALS(h.yMode(), Histogram::YMode::Frequencies);
+  }
+
+  void test_yMode_initialized_by_setFrequencyVariances() {
+    Histogram h(Points(2));
+    h.setFrequencyVariances(2);
+    TS_ASSERT_EQUALS(h.yMode(), Histogram::YMode::Frequencies);
+  }
+
+  void test_yMode_cannot_be_changed_by_setters() {
+    Histogram countHist(Points(1), Counts(1));
+    TS_ASSERT_THROWS(countHist.setFrequencies(1), std::logic_error);
+    TS_ASSERT_THROWS(countHist.setFrequencyVariances(1), std::logic_error);
+    TS_ASSERT_THROWS(countHist.setFrequencyStandardDeviations(1),
+                     std::logic_error);
+    Histogram freqHist(Points(1), Frequencies(1));
+    TS_ASSERT_THROWS(freqHist.setCounts(1), std::logic_error);
+    TS_ASSERT_THROWS(freqHist.setCountVariances(1), std::logic_error);
+    TS_ASSERT_THROWS(freqHist.setCountStandardDeviations(1), std::logic_error);
+  }
```

Comment:
Good, I would probably have broken this up into more individual test methods to test each scenario.